server/src/main/java/org/opensearch/action/main/MainResponse.java

```diff
@@ -102,7 +102,11 @@ public Build getBuild() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(nodeName);
-        Version.writeVersion(version, out);
+        if (out.getVersion().before(Version.V_1_0_0)) {
+            Version.writeVersion(LegacyESVersion.V_7_10_2, out);
+        } else {
+            Version.writeVersion(version, out);
+        }
```

Comment:
Is there a way we could add a test using legacy client?

Reply:
BWC testing does this.

Reply:
Seems like that would be too late to catch bugs in this code. For a unit test I would imagine a simple "expects a legacy version vs. current version when calling `writeTo`.